dts/arm/ambiq/ambiq_apollo3p_blue.dtsi

```diff
@@ -0,0 +1,227 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#include <arm/armv7-m.dtsi>
+#include <mem.h>
+#include <freq.h>
+#include <zephyr/dt-bindings/i2c/i2c.h>
+#include <zephyr/dt-bindings/gpio/gpio.h>
+
+/ {
+	clocks {
+		uartclk: apb-pclk {
+			compatible = "fixed-clock";
+			clock-frequency = <24000000>;
+			#clock-cells = <0>;
+		};
+	};
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: cpu@0 {
+			compatible = "arm,cortex-m4f";
+			reg = <0>;
+		};
+	};
+
+	/* TCM */
+	tcm: tcm@10000000 {
+		compatible = "zephyr,memory-region";
+		reg = <0x10000000 0x10000>;
+		zephyr,memory-region = "ITCM";
+	};
+
+	/* SRAM */
+	sram0: memory@10010000 {
+		compatible = "mmio-sram";
+		reg = <0x10010000 0xb0000>;
+	};
+
+	soc {
+		flash: flash-controller@c000 {
+			compatible = "ambiq,flash-controller";
+			reg = <0x0000C000 0x1f4000>;
+
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			/* MRAM region */
+			flash0: flash@c000 {
+				compatible = "soc-nv-flash";
+				reg = <0x0000C000 0x1f0000>;
+			};
+		};
+
+		pwrcfg: pwrcfg@40021000 {
+			compatible = "ambiq,pwrctrl";
+			reg = <0x40021000 0x400>;
+			#pwrcfg-cells = <2>;
+		};
+
+		stimer0: stimer@40008800 {
+			compatible = "ambiq,stimer";
+			reg = <0x40008800 0x80>;
+			interrupts = <23 0>;
+			status = "okay";
+		};
+
+		counter0: counter@40008000 {
+			compatible = "ambiq,counter";
+			reg = <0x40008000 0x80>;
+			interrupts = <14 0>;
+			status = "disabled";
+		};
+
+		uart0: uart@4001c000 {
+			compatible = "ambiq,uart", "arm,pl011";
+			reg = <0x4001c000 0x1000>;
+			interrupts = <15 0>;
+			interrupt-names = "UART0";
+			status = "disabled";
+			clocks = <&uartclk>;
+			ambiq,pwrcfg = <&pwrcfg 0x8 0x80>;
+		};
+		uart1: uart@4001d000 {
+			compatible = "ambiq,uart", "arm,pl011";
+			reg = <0x4001d000 0x1000>;
+			interrupts = <16 0>;
+			interrupt-names = "UART1";
+			status = "disabled";
+			clocks = <&uartclk>;
+			ambiq,pwrcfg = <&pwrcfg 0x8 0x100>;
+		};
+
+		iom0: iom@40050000 {
```

Comment:
Just double checking the addresses for the IOM and maybe the stimer.

I may be looking at the wrong datasheet/header file but in `AmbiqSuite_R3.0.0/CMSIS/AmbiqMicro/Include/apollo3p.h` I see no mention an STIMER and also the IOM addresses appear to be different.

```
#define CTIMER_BASE                 0x40008000UL
#define IOM0_BASE                   0x50004000UL
#define IOM1_BASE                   0x50005000UL
#define IOM2_BASE                   0x50006000UL
#define IOM3_BASE                   0x50007000UL
#define IOM4_BASE                   0x50008000UL
#define IOM5_BASE                   0x50009000UL
```

Reply:
From how they use it in their HAL and examples, the stimer is the same as the ctimer on Apollo3. On Apollo4 they have separate hardware for it.

Good catch on the i2c registers. I was probably looking at the apollo4 data sheet when making the device tree.